Mage.Sets/src/mage/cards/g/GerrardsHourglassPendant.java

```diff
@@ -0,0 +1,129 @@
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.keyword.FlashAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterPermanentCard;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.card.PutIntoGraveFromBattlefieldThisTurnPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.watchers.common.CardsPutIntoGraveyardWatcher;
+
+import java.util.UUID;
+
+public final class GerrardsHourglassPendant extends CardImpl {
+
+    public GerrardsHourglassPendant(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{1}");
+        addSuperType(SuperType.LEGENDARY);
+
+        // Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        // If a player would begin an extra turn, that player skips that turn instead.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GerrardsHourglassPendantSkipExtraTurnsEffect()));
+
+        // {4}, {T}, Exile Gerrard's Hourglass Pendant: Return to the battlefield tapped all artifact, creature,
+        // enchantment, and land cards in your graveyard that were put there from the battlefield this turn.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new GerrardsHourglassPendantReanimateEffect(), new GenericManaCost(4));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new ExileSourceCost());
+        ability.addWatcher(new CardsPutIntoGraveyardWatcher());
+        this.addAbility(ability);
+    }
+
+    private GerrardsHourglassPendant(final GerrardsHourglassPendant card) {
+        super(card);
+    }
+
+    @Override
+    public GerrardsHourglassPendant copy() {
+        return new GerrardsHourglassPendant(this);
+    }
+}
+
+class GerrardsHourglassPendantSkipExtraTurnsEffect extends ReplacementEffectImpl {
+
+    public GerrardsHourglassPendantSkipExtraTurnsEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Detriment);
+        staticText = "If a player would begin an extra turn, that player skips that turn instead";
+    }
+
+    public GerrardsHourglassPendantSkipExtraTurnsEffect(final GerrardsHourglassPendantSkipExtraTurnsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public GerrardsHourglassPendantSkipExtraTurnsEffect copy() {
+        return new GerrardsHourglassPendantSkipExtraTurnsEffect(this);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        Player player = game.getPlayer(event.getPlayerId());
+        MageObject sourceObject = game.getObject(source);
+        if (player != null && sourceObject != null) {
+            game.informPlayers(sourceObject.getLogName() + ": Extra turn of " + player.getLogName() + " skipped");
+        }
```

Comment:
I think this should log the fact that the extra turn was skipped even if the source object (the pendant) is no longer on the battlefield.

Reply:
Is it possible for the object to not be on the battlefield when the replacement effect happens given that the interaction doesn't use the stack? It seems like checking that `player` and `sourceObject` aren't null is just as a means of good practice before trying to call `getLogName()` more so than to accommodate edge cases